pkg/extract/tar.go

```diff
@@ -22,10 +22,23 @@ type link struct {
 	newName  string
 }
 
-func TarFile(reader io.Reader, destDir string, overwrite bool) error {
+func TarFile(r io.Reader, destDir string, overwrite bool) error {
 	var links []*link
 
 	startTime := time.Now()
+	peekReader := &peekReader{reader: r}
```

Comment:
```suggestion
	peekReader := bufio.NewReaderSize(r,8)
```
then peekReader.Peek(8) gets you the magic bytes